build-support/docker/centos6/Dockerfile

```diff
@@ -4,18 +4,33 @@
 # Use centos6 for compatibility with glibc 2.12.
 FROM centos:6
 
-# Install python 2.7.13, (more) modern gcc, and a JDK.
+# Install a (more) modern gcc, and a JDK.
+# Also install sqlite-devel because python 3 errors on startup if not -- see
+# https://stackoverflow.com/a/39907500/2518889.
 RUN yum -y update
+# TODO: figure out why this needs to be installed first for /usr/bin/scl to work!
 RUN yum install -y centos-release-scl
-RUN yum install -y \
-  devtoolset-7-gcc \
-  devtoolset-7-gcc-c++ \
+RUN yum install -y devtoolset-7-gcc{,-c++} \
   git \
   java-1.8.0-openjdk-devel \
-  python27 \
-  libffi \
-  libffi-devel \
-  openssl-devel
+  libffi{,-devel} \
+  openssl-devel \
+  sqlite-devel
+
+# Install the desired versions of python 2 and 3.
+ARG PYTHON_2_VERSION=2.7.13
+ARG PYTHON_3_VERSION=3.6.8
+# NB: PYENV_ROOT must be set in the environment for `pyenv install` to work, otherwise it will
+# mysteriously fail to find the `install` command.
```

Comment:
Is this mysterious failure due to not running a in new shell, so it picks up the path changes?

Also, this might be an easier way to install pyenv: 

https://github.com/pyenv/pyenv-installer

Reply:
In the `bin/` directory of the pyenv repo, there's all these other commands, just not `install`, and I didn't dive long enough to figure out why. I'll use pyenv-installer.

Reply:
No, it's not due to running in a new shell. It's because Pyenv's code requires `PYENV_ROOT` to function properly and assumes it's set.